benchmarking/smartseq2/compare_matrix/Compare_data_matrix.R

```diff
@@ -0,0 +1,380 @@
+    # log2 transformation
+    mat1.log <- takelog2(mat1)
+    mat2.log <- takelog2(mat2)
+  } else{
+    mat1.log <- mat1
+    mat2.log <- mat2
+  }
+  #mcor <-  cor(mat1.log[,-c(1:2)],mat2.log[,-c(1:2)])
+  stat.cor <- RunCorrTest(mat1.log[,-c(1:2)], mat2.log[,-c(1:2)])
+  return(stat.cor)
+}
+# summary fold changes by gene's biotype
+# foldchanges: median or mean of FC cross dataset
+# genes: gene annotation from gft file
+# threshold: a cut-off to select significant foldchange
+# values.
+summaryFoldChanges <- function(foldchanges, genes, threshold) {
+  upIDs <- names(which(foldchanges > threshold))
+  downIDs <- names(which(foldchanges < -threshold))
+  # parse up- down- genes
+  upGene <- subset(genes, genes$gene_id %in% upIDs)
+  downGene <- subset(genes, genes$gene_id %in% downIDs)
+  upGene <- cbind('FC' = rep('UP', nrow(upGene)), upGene)
+  downGene <- cbind('FC' = rep('DN', nrow(downGene)), downGene)
+  # put together
+  fcGene <- rbind(upGene, downGene)
+  # summary up- down- FC
+  fc_tb <- table(fcGene$FC, fcGene$gene_type)
+  fc_tb['DN', ] <- (-1) * fc_tb['DN', ]
+  return(fc_tb)
+}
+# Run SNN-Cliq to generate SNN graphic and
+# then cluster cells based on graph to generate cluster/group
+# then visualize the SNN clustering result in PCA
+RunSNNCluster <- function(datamatrix) {
+  mat.log <- takelog2(datamatrix)
+  # First run randome PCA then extract top 2 PCs for visualization
+  mat.pca <- rpca(mat.log[,-c(1:2)], scale = T)
+  PCs <- mat.pca$rotation[, 1:2]
+  # Run SNN graph
+  grps <- buildSNNGraph(mat.log[,-c(1:2)])
+  # extract memebership
+  clusters <- cluster_fast_greedy(grps)
+  fc <-  membership(clusters)
+  out <-
+    data.frame('PC1' = PCs[, 1],
+               'PC2' = PCs[, 2],
+               'membership' = as.factor(fc))
+  return(out)
+}
+
+# inputs
+option_list <- list(
+  make_option(
+    "--matrix1",
+    type = "character",
+    default = NULL,
+    help = "data matrix file name",
+    metavar = "character"
+  ),
+  make_option(
+    "--matrix2",
+    type = "character",
+    default = NULL,
+    help = "updated data matrix file name",
+    metavar = "character"
+  ),
+  make_option(
+    "--gtf_file",
+    type = "character",
+    default = NULL,
+    help = "gtf annotation file",
+    metavar = "character"
+  ),
+  make_option(
+    "--output_prefix",
+    type = "character",
+    default = NULL,
+    help = "output prefix",
+    metavar = "character"
+  )
+)
+opt_parser <- OptionParser(option_list = option_list)
+opt <- parse_args(opt_parser)
+matrixfile1 <- opt$matrix1
+matrixfile2 <- opt$matrix2
+gtf_file <-  opt$gtf_file
+output_name <- opt$output_prefix
+# color palette
+palette(c("#00AFBB", "#E7B800"))
+# load data matrix
+mat1 <- read.csv(matrixfile1)
+mat2 <- read.csv(matrixfile2)
+# calculate folder changes
+fc <- foldchanges(mat1, mat2)
+# calucate mean or median FC cross samples
+fc.dd <- fc[,-c(1:2)]
+fc.m <- apply(fc.dd, 1, median)
+# boxplot
+# Box plot (bp)
+bxp <- ggviolin(
```

Comment:
it is not really because the scale. it probably due to the narrow distribution of foldchanges and most of data point clustered around 0.  But I will look into this suggestion later